opm/material/fluidmatrixinteractions/EclEpsScalingPoints.hpp

```diff
@@ -156,105 +136,38 @@ struct EclEpsScalingPointsInfo
-        const TableContainer&  sgofTables = tables.getSgofTables();
-        const TableContainer& slgofTables = tables.getSlgofTables();
-        const TableContainer&  swfnTables = tables.getSwfnTables();
-        const TableContainer&  sgfnTables = tables.getSgfnTables();
-        const TableContainer&  sof3Tables = tables.getSof3Tables();
-        const TableContainer&  sof2Tables = tables.getSof2Tables();
-
-
-        bool hasWater = eclState.runspec().phases().active(Phase::WATER);
-        bool hasGas = eclState.runspec().phases().active(Phase::GAS);
-        bool hasOil = eclState.runspec().phases().active(Phase::OIL);
-
-        if (int(hasWater) + int(hasGas) + int(hasOil) == 1) {
-            return;
-        } else if (!hasWater) {
-            Swl = 0.0;
-            Swu = 0.0;
-            Swcr = 0.0;
-            bool family1 = (!sgofTables.empty() || !slgofTables.empty());
-            bool family2 = !sgfnTables.empty() && !sof2Tables.empty();
-            if (family1) {
-                if (!sgofTables.empty())
-                    extractUnscaledSgof_(sgofTables.getTable<SgofTable>(satRegionIdx));
-                else {
-                    assert(!slgofTables.empty());
-                    extractUnscaledSlgof_(slgofTables.getTable<SlgofTable>(satRegionIdx));
-                }
-            } else if (family2) {
-                extractUnscaledSgfn_(sgfnTables.getTable<SgfnTable>(satRegionIdx));
-                extractUnscaledSof2_(sof2Tables.getTable<Sof2Table>(satRegionIdx));
-            }
-            else {
-                throw std::domain_error("No valid saturation keyword family specified");
-            }
-            return;
-        } else if (!hasGas) {
-            Sgl = 0.0;
-            Sgu = 0.0;
-            Sgcr = 0.0;
-            bool family1 = !swofTables.empty();
-            bool family2 = !swfnTables.empty() && !sof2Tables.empty();
-            if (family1) {
-                extractUnscaledSwof_(swofTables.getTable<SwofTable>(satRegionIdx));
-            } else if (family2) {
-                extractUnscaledSwfn_(swfnTables.getTable<SwfnTable>(satRegionIdx));
-                extractUnscaledSof2_(sof2Tables.getTable<Sof2Table>(satRegionIdx));
-            }
-            else {
-                throw std::domain_error("No valid saturation keyword family specified");
-            }
-            return;
+        this->Swl = rtep.connate.water[satRegionIdx];
+        this->Sgl = rtep.connate.gas  [satRegionIdx];
+
+        this->Swcr  = rtep.critical.water       [satRegionIdx];
+        this->Sgcr  = rtep.critical.gas         [satRegionIdx];
+        this->Sowcr = rtep.critical.oil_in_water[satRegionIdx];
+        this->Sogcr = rtep.critical.oil_in_gas  [satRegionIdx];
+        if (family == SatFuncControls::KeywordFamily::Family_I) {
+            // Hack.  Papers over unknown problems elsewhere.
+            this->Sogcr += this->Swl;
         }
```

Comment:
Could you add an "address" to elsewhere in case this HACK is long lived. It would make it possible to others to investigate this later.

Reply:
The trouble is I don't know where the issue is.  It is likely to be a combination of problems in multiple files.